### DISCUSSION TITLE
Help splitting an image into tiles

### DISCUSSION THREAD

**Discussion by OrionNavattan** [ImageMagick/ImageMagick#6244] [Help]
### ImageMagick version

7.1.1-6

### Operating system, version and so on

macOS Ventura 13.3.1

### Description

I have several thousand 256x256 pixel PNGs that I would like to split into smaller 128x128 tiles, and my attempt at using the crop command has gone nowhere. Based on what I've read on the command line options webpage and elsewhere, I came up with this:

```convert "inputfile" +repage -crop 128x128+0+0@! +repage "outputfile"```

However, it's failing spectacularly: instead of 4 128x128 images, the above generates **16384 2x2** images instead. Removing the "@" symbol from the geometry does produce the expected result, but only for the upper left of the image. Is there something wrong with my arguments, or have I stumbled across a bug?

**Comment by snibgo** [comment]
You use IM v7, but with `convert` which uses the (very old) v6 syntax. I suggest you use `magick` instead to get the much better v7 syntax (and not `magick convert`, which would also use the old syntax).

I suggest one of these methods:
```
magick "inputfile" +repage -crop 2x2@ +repage "outputfile"
```
This chops the input into 2 pieces horizontally and 2 pieces vertically.

Alternatively:
```
magick "inputfile" +repage -crop 128x128 +repage "outputfile"
```
This splits the input into rows that are 128 pixels high, and splits each row into pieces that are 128 pixels wide.

So the effect in your case is the same. 

Note that neither of these methods use an offset like `+0+0`.
